Adding Threat Matrix and additional Threats to Section 3.3 and Section 3.2 respectively of draft-ietf-dult-threat-model

shaileshprabhu95 opened the pull request:
This PR adds:

1. Additional threats to Section 3.2
2. Introducing a Threat Matrix in Section 3.3

maggiedelano reviewed (changes requested):
Thanks for your PR!

I made some minor commits, fixing the trailing white space, and also truncating the column headers because they were too long to fit the HTML view appropriately.

I think a few additions would be helpful:
1. Adding the low/medium/high descriptions to the text since they were removed from the headers
2. We should discuss the profile of each attack in a bit more detail in each of the paragraphs now that we have the table. This reasoning will also help me and others assess whether each cell in the matrix is appropriate or not.
3. Can you define what you mean by mitigation available? (And if it makes sense, the definitions of any of the other columns)

If you can take a first pass at this, that would be great, and I can help update further as needed.

shaileshprabhu95 commented:
Hi Maggie,

Thank you for the review. I have updated this PR to incorporate your feedback. The revisions include:

1. Adding descriptions for the impact, likelihood, feasibility, and risk levels that were removed from the Threat Matrix headers.
2. Expanding the attack profiles with additional details on how each attack works, its feasibility, and its impact.
3. Defining "Mitigation Available" and providing definitions for the other columns as well.

Please review the updated PR and let me know if any further refinements are needed. Thanks! @maggiedelano

shaileshprabhu95 commented:
Hi Maggie,

Thanks for reviewing it further and adding suggestions. I have addressed your review comments in the latest commit with the below modifications:

1. I've rephrased the risk statements to explicitly consider **impact**, **likelihood**, and **feasibility** in a balanced manner.
2. The attacks with the same inputs have the same risk score. Each attack now follows the below logical pattern:
    - High Impact + High Likelihood + Easy Feasibility → High Risk
    - High Impact + Medium Likelihood + Moderate/Hard Feasibility → Medium Risk
    - Medium Impact + Medium Likelihood + Moderate Feasibility → Medium Risk
3. I've moved the threat matrix and table to the beginning of the section.

Please review and let me know if you need further refinements. Thanks! @maggiedelano

maggiedelano commented:
Fixed some minor formatting issues, merging now.
